### PR TITLE
Add GC predicate functions

### LINKED ISSUE #1158: Predicate functions for General_Category
For use cases in which we want to request the General_Category property value for a given code point (ex: iterating through text per-code point), we should provide predicate functions that allow the user to query whether the returned property value is included in the multi-category values.

The [General_Category property value](https://unicode.org/reports/tr44/#General_Category_Values) describes the 30 "simple" property values (Uppercase_Letter, Lowercase_Letter, ...), which are the what is returned for the first API described above. 

The multi-category values (also described as "derived" values) are logically set unions of other General_Category values (Cased_Letter, Letter, Mark, ...). Just like in ICU, we need predicate functions that correspond to each multi-category value (`is_cased_letter(...) -> bool`, `is_letter(...) -> bool`, ...) that return a boolean when the input belongs to the corresponding logical set.

### PR DESCRIPTION
**Pull request by echeran**
Fixes #1158

### CODE REVIEW

File: components/properties/src/maps.rs [hunk 1]
```diff
@@ -216,3 +216,204 @@ where
...
+/// let data_struct = payload.get();
+/// let gc = &data_struct.code_point_trie;
+/// 
+/// assert_eq!(gc.get('A' as u32), GeneralSubcategory::UppercaseLetter);
+/// assert!(maps::is_cased_letter(&gc, 'A' as u32));
+/// ```
+pub fn is_cased_letter(gc_map: &CodePointTrie<GeneralSubcategory>, code_point: u32) -> bool {
+    let gc_val = gc_map.get(code_point);
+    0 != (gc_val as u32) & GeneralCategory::CasedLetter.0
+}
```
Comment (sffc) [inline components/properties/src/maps.rs:251]: Two other ways:

```rust
impl GeneralSubcategory {
    pub fn is_contained(gc: GeneralCategory) -> bool {
        // do the test
    }
}
```

```rust
impl GeneralCategory {
    pub fn contains(gsc: GeneralSubcategory) -> bool {
        // opposite of above
    }
}
```

File: components/properties/src/maps.rs [hunk 2]
```diff
@@ -216,3 +216,204 @@ where
...
+///         .expect("The data should be valid");
+/// let data_struct = payload.get();
+/// let gc = &data_struct.code_point_trie;
+/// 
+/// assert_eq!(gc.get('𝔅' as u32), GeneralSubcategory::UppercaseLetter);  // U+1D505 MATHEMATICAL FRAKTUR CAPITAL B
+/// assert!(maps::is_letter(&gc, '𝔅' as u32));
+/// ```
+pub fn is_letter(gc_map: &CodePointTrie<GeneralSubcategory>, code_point: u32) -> bool {
+    let gc_val = gc_map.get(code_point);
+    0 != (gc_val as u32) & GeneralCategory::Letter.0
```
Comment (sffc) [inline components/properties/src/maps.rs:275]:
```suggestion
    0 != (1 << (gc_val as u32)) & GeneralCategory::Letter.0
```

**Review by sffc — CHANGES_REQUESTED** [review]

File: components/properties/src/props.rs [hunk 3]
```diff
@@ -240,6 +240,40 @@ impl GeneralCategory {
...
+    /// assert_eq!(gc.get('0' as u32), GeneralSubcategory::DecimalNumber);
+    /// assert!(GeneralCategory::Number.contains(gc.get('0' as u32)));
+    /// assert_eq!(gc.get('(' as u32), GeneralSubcategory::OpenPunctuation);
+    /// assert!(GeneralCategory::Punctuation.contains(gc.get('(' as u32)));
+    /// assert_eq!(gc.get('✓' as u32), GeneralSubcategory::OtherSymbol);
+    /// assert!(GeneralCategory::Symbol.contains(gc.get('✓' as u32)));
+    /// assert_eq!(gc.get(' ' as u32), GeneralSubcategory::SpaceSeparator);
+    /// assert!(GeneralCategory::Separator.contains(gc.get(' ' as u32)));
+    /// assert_eq!(gc.get(0xE007F), GeneralSubcategory::Format);  // U+E007F CANCEL TAG
+    /// assert!(GeneralCategory::Other.contains(gc.get(0xE007F)));
```
Comment (sffc) [inline components/properties/src/props.rs:272]: Nit: Add some more spacing here to improve readability, and also add some negative containment tests.

Also please add a test for a Letter that is not CasedLetter.

```suggestion
    /// assert_eq!(gc.get('A' as u32), GeneralSubcategory::UppercaseLetter);
    /// assert!(GeneralCategory::CasedLetter.contains(gc.get('A' as u32)));
    ///
    /// // U+1D505 MATHEMATICAL FRAKTUR CAPITAL B
    /// assert_eq!(gc.get('𝔅' as u32), GeneralSubcategory::UppercaseLetter);
    /// assert!(GeneralCategory::Letter.contains(gc.get('𝔅' as u32)));
    ///
    /// // U+0301 COMBINING ACUTE ACCENT
    /// assert_eq!(gc.get(0x0301), GeneralSubcategory::NonspacingMark);
    /// assert!(GeneralCategory::Mark.contains(gc.get(0x0301)));
    /// assert!(!GeneralCategory::Letter.contains(gc.get(0x0301)));
    ///
    /// assert_eq!(gc.get('0' as u32), GeneralSubcategory::DecimalNumber);
    /// assert!(GeneralCategory::Number.contains(gc.get('0' as u32)));
    /// assert!(!GeneralCategory::Mark.contains(gc.get('0' as u32)));
    ///
    /// assert_eq!(gc.get('(' as u32), GeneralSubcategory::OpenPunctuation);
    /// assert!(GeneralCategory::Punctuation.contains(gc.get('(' as u32)));
    /// assert!(!GeneralCategory::Number.contains(gc.get('(' as u32)));
    ///
    /// assert_eq!(gc.get('✓' as u32), GeneralSubcategory::OtherSymbol);
    /// assert!(GeneralCategory::Symbol.contains(gc.get('✓' as u32)));
    /// assert!(!GeneralCategory::Punctuation.contains(gc.get('✓' as u32)));
    ///
    /// assert_eq!(gc.get(' ' as u32), GeneralSubcategory::SpaceSeparator);
    /// assert!(GeneralCategory::Separator.contains(gc.get(' ' as u32)));
    /// assert!(!GeneralCategory::Symbol.contains(gc.get(' ' as u32)));
    ///
    /// // U+E007F CANCEL TAG
    /// assert_eq!(gc.get(0xE007F), GeneralSubcategory::Format);
    /// assert!(GeneralCategory::Other.contains(gc.get(0xE007F)));
    /// assert!(!GeneralCategory::Separator.contains(gc.get(0xE007F)));
```

**Review by sffc — APPROVED** [review]
LGTM with a suggestion for better documentation

File: components/properties/src/maps.rs (hunk 2 above)
Comment (echeran) [reply components/properties/src/maps.rs]: Done before a previous review.

File: components/properties/src/maps.rs (hunk 1 above)
Comment (echeran) [reply components/properties/src/maps.rs]: Done before a previous review. Took the second approach.

**Review by sffc — APPROVED** [review]

**Pull request merged**